src/OpenTelemetryEngine/Traces/FunctionsActivityBuilder.cs

```diff
@@ -0,0 +1,81 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System.Diagnostics;
+using OpenTelemetryEngine.Constants;
+using OpenTelemetryEngine.ResponseObjects;
+
+namespace OpenTelemetryEngine.Traces
+{ 
+    public class FunctionsActivityBuilder
+    {
+        public static ActivitySource sourceInternal = new ActivitySource("AzureFunctionsInternal");
+        public static ActivitySource source = new ActivitySource(OpenTelemetryModuleConstants.ActivitySourceName);
+
+        private static Dictionary<string, Activity> internalActivitiesByInvocationId = new Dictionary<string, Activity>();
+
+        public static FunctionsActivityResponse StartInternalActivity(string invocationId, string traceParent, string traceState) 
```

Comment:
It looks like this method can be invoked from multiple threads concurrently. If this is indeed the case, we probably need to protect `internalActivitiesByInvocationId`, as I believe `Dictionary` is not thread-safe.

Reply:
Updated to use ConcurrentDictionary, and added guards and comments about add/remove failure